lib/st_codebreaker/matcher.rb

```diff
@@ -0,0 +1,37 @@
+module StCodebreaker
+  class Matcher
+    def initialize(secret, guess)
+      @secret = secret
+      @guess = guess
+    end
+
+    def exact_match
+      (0..3).inject(0) do |count, index|
+        count + (exact_match?(index) ? 1 : 0)
+      end
+    end
+
+    def number_match
+      total_match - exact_match
+    end
+
+    def total_match
+      secret = @secret.split('')
+      @guess.split('').inject(0) do |count, n|
+        count + (delete_first(secret, n) ? 1 : 0)
```

Comment:
It is not good to iterate array twice (line 10). Can you do it in one loop somehow?